arcjet/index.ts

```diff
@@ -554,20 +555,34 @@ export function validateEmail(
         context: ArcjetContext,
         { email }: ArcjetRequestDetails & { email: string },
       ): Promise<ArcjetRuleResult> {
-        if (await analyze.isValidEmail(context, email, analyzeOpts)) {
+        const result = await analyze.isValidEmail(context, email, emailOpts);
+        console.log(result);
+        if (result.validity === "valid") {
           return new ArcjetRuleResult({
             ttl: 0,
             state: "RUN",
             conclusion: "ALLOW",
             reason: new ArcjetEmailReason({ emailTypes: [] }),
           });
         } else {
+          const typedEmailTypes = result.blocked.filter(
+            (type): type is ArcjetEmailType => {
+              return (
+                type === "FREE" ||
+                type === "DISPOSABLE" ||
+                type === "NO_MX_RECORDS" ||
+                type === "NO_GRAVATAR" ||
+                type === "INVALID"
+              );
+            },
```

Comment:
This should be a named function near the top of the file. We don't want to be creating a new function on every execution.